src/crawlers/cheerio_crawler.js

```diff
@@ -530,6 +570,29 @@ class CheerioCrawler extends BasicCrawler {
         );
     }
 
+    /**
+     * @param {CrawlingContext} crawlingContext
+     * @ignore
+     * @protected
+     * @internal
+     */
+    async _handleNavigation(crawlingContext) {
+        const gotoOptions = { ...this.defaultGotoOptions };
+        if (this.prepareRequestFunction) await this.prepareRequestFunction(crawlingContext);
+        await this._executeHooks(this.preNavigationHooks, crawlingContext, gotoOptions);
```

Comment:
Maybe we should show a warning if someone uses `prepareRequestFunction` together with `preNavigationHooks`? Or even deprecate the old method?

Reply:
I see the comment now, yes, deprecation makes sense.